static/js/components/material/Snackbar.js

```diff
@@ -0,0 +1,52 @@
+// @flow
+import React from "react"
+
+import { MDCSnackbar } from "@material/snackbar/dist/mdc.snackbar"
+
+import type { SnackbarState } from "../../reducers/ui"
+
+type SnackbarProps = {
+  snackbar: ?SnackbarState
+}
+
+export default class Snackbar extends React.Component<*, *> {
+  snackbar = null
+  snackbarRoot = null
+
+  props: SnackbarProps
+
+  componentDidMount() {
+    this.snackbar = new MDCSnackbar(this.snackbarRoot)
+  }
+
+  componentWillReceiveProps(nextProps: SnackbarProps) {
+    if (this.props.snackbar !== nextProps.snackbar) {
```

Comment:
This will compare identity, not value. Is there something in ramda for deep equality?

Reply:
Ok, I'll need to add in something to guarantee object uniqueness per dispatch of `setSnackbarMessage` too.

Reply:
I added a sequential id to the state, this should guarantee the above.